### PR TITLE
Update release script + release `0.8.0-alpha.2`

### PR DESCRIPTION
**Pull request by jprochazk**
### What

- Fixed many issues that appeared during the publishing process
- Improved the output of `scripts/ci/crates.py`
- `scripts/ci/crates.py` now uses the `crates.io` API to check directly if the version we're trying to publish has already been published. This is much faster than waiting for an error from `cargo publish`.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2609) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2609)
- [Docs preview](https://rerun.io/preview/pr%3Ajan%2Frelease-0.8.0-alpha/docs)
- [Examples preview](https://rerun.io/preview/pr%3Ajan%2Frelease-0.8.0-alpha/examples)

### CODE REVIEW

**Comment by emilk** [comment]
I've published and updated to `egui_tiles` 0.2.0

**Comment by jprochazk** [comment]
Successfully published everything 🎉

File: crates/re_log_types/Cargo.toml [hunk 1]
```diff
@@ -60,6 +60,7 @@ thiserror.workspace = true
 time = { workspace = true, features = ["formatting", "macros"] }
 typenum = "1.15"
 uuid = { version = "1.1", features = ["serde", "v4", "js"] }
+half.workspace = true
```
Comment (emilk) [inline crates/re_log_types/Cargo.toml:63]: sort! and maybe add a note about why it is here even though it is not a direct dependency

File: scripts/ci/crates.py [hunk 2]
```diff
@@ -265,7 +266,46 @@ def version(dry_run: bool, bump: Bump) -> None:
...
+
+def publish_crate(crate: Crate, token: str, version: str, env: Dict[str, Any]) -> None:
+    package = crate.manifest["package"]
+    name = package["name"]
+    crate_version = crate.manifest["package"].get("version") or version
+    if "workspace" in crate_version:
+        crate_version = version
+
+    if is_already_uploaded(crate_version, crate.manifest["package"]["name"]):
+        print(f"{Fore.GREEN}Skipped{Fore.RESET} {Fore.BLUE}{name}{Fore.RESET}")
```
Comment (emilk) [inline scripts/ci/crates.py:298]: should probably print why it was skipped

Comment (emilk) [inline scripts/ci/crates.py:298]:
```suggestion
        print(f"{Fore.GREEN}Already published{Fore.RESET} {Fore.BLUE}{name}{Fore.RESET}")
```

File: scripts/ci/crates.py [hunk 3]
```diff
@@ -280,9 +320,9 @@ def publish(dry_run: bool, token: str) -> None:
     ctx.finish(dry_run)
 
     if not dry_run:
+        env = {**os.environ.copy(), "RERUN_IS_PUBLISHING": "yes", "IS_IN_RERUN_WORKSPACE": "yes"}
```
Comment (emilk) [inline scripts/ci/crates.py:323]: I don't think we should set `IS_IN_RERUN_WORKSPACE` here. That is meant for detecting a developer (us) working locally.

**Review by emilk — APPROVED** [review]
`scripts/publish_crates.sh` also builds the web viewer before publishing - is this now done implicitly on the CI?

**Comment by jprochazk** [comment]
> scripts/publish_crates.sh also builds the web viewer before publishing - is this now done implicitly on the CI?

Nothing related to publishing is done on CI right now, CI changes will go in a follow-up PR once we resolve the rate limiting issue.

**Comment by emilk** [comment]
Seems like `scripts/publish_crates.sh` should be updated to call `scripts/ci/crates.py`

File: crates/rerun_c/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# rerun_c
+
+Part of the [`rerun`](https://github.com/rerun-io/rerun) family of crates.
+
+[![Latest version](https://img.shields.io/crates/v/rerun_c.svg)](https://crates.io/crates/rerun_c)
+[![Documentation](https://docs.rs/rerun_c/badge.svg)](https://docs.rs/rerun_c)
+![MIT](https://img.shields.io/badge/license-MIT-blue.svg)
+![Apache](https://img.shields.io/badge/license-Apache-blue.svg)
+
+Rerun C bindings.
```
Comment (emilk) [inline crates/rerun_c/README.md:10]:
```suggestion
Rerun C bindings. Not ready to be used yet.
```

**Comment by jprochazk** [comment]
> Seems like scripts/publish_crates.sh should be updated to call scripts/ci/crates.py

We could, or just add the remaining functionality into `scripts/ci/crates.py`. I think that should be in a follow-up PR. I'll also need a bit of help understanding what it does, probably during standup

**Pull request merged**